### PR TITLE
[RHDM-645] "java.security.AccessControlException: WFSM000001: Permission check failed" when enabled security-manager

### PR DESCRIPTION
**Pull request by elguardian**
avoiding common pool in parallel streams so it checks properly with the security manager

### CODE REVIEW

File: drools-compiler/src/main/java/org/drools/compiler/builder/impl/KnowledgeBuilderImpl.java [hunk 1]
```diff
@@ -1397,19 +1398,26 @@ private void compileRulesLevel(PackageDescr packageDescr, PackageRegistry pkgReg
...
-                                this.results.addAll(results);
+                                    this.results.addAll(results);
+                                }
                             }
-                        }
-                    });
+                        })
+                ).get();
+            } catch (InterruptedException | ExecutionException e) {
+                logger.error("An error has occurred during rules compilation", e);
```
Comment (baldimir) [inline drools-compiler/src/main/java/org/drools/compiler/builder/impl/KnowledgeBuilderImpl.java]: Not sure, if we want the error just logged. What do you think @mariofusco?

Comment (mariofusco) [reply drools-compiler/src/main/java/org/drools/compiler/builder/impl/KnowledgeBuilderImpl.java]: @elguardian @baldimir No, in this way the error will be swallowed. It needs to be rethrown (wrapped in a RuntimeException if necessary).

**Comment by elguardian** [comment]
@baldimir any progress with the review ? a pushed a new commit the changes asked

**Comment by elguardian** [comment]
ok to test

**Review by baldimir — APPROVED** [review]
Sorry, I missed the new commit. Approving.

**Pull request merged**